### PR TITLE
Browserify => Webpack

### PR DESCRIPTION
**Pull request by aputinski**
`config.src.scripts.toolkit` is now bundled using webpack.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -19,6 +18,7 @@ var sass = require('gulp-sass');
 var source = require('vinyl-source-stream');
 var streamify = require('gulp-streamify');
```
Comment (LukeAskew) [inline gulpfile.js]: `source` and `streamify` can be removed and uninstalled since they are no longer used.

File: gulpfile.js [hunk 2]
```diff
@@ -77,6 +76,9 @@ gulp.task('styles', ['styles:fabricator', 'styles:toolkit']);
 
 
 // scripts
+var webpackCompiler = webpack(require('./webpack.config')(config));
```
Comment (LukeAskew) [inline gulpfile.js]: Can this be moved to the `var` block at the top, next to the `webpack` var delcaration? Just a matter style preference
.

File: gulpfile.js [hunk 3]
```diff
@@ -84,16 +86,17 @@ gulp.task('scripts:fabricator', function () {
...
-		.on('error', function (error) {
-			gutil.log(gutil.colors.red(error));
-			this.emit('end');
-		})
-		.pipe(source('toolkit.js'))
-		.pipe(gulpif(!config.dev, streamify(uglify())))
-		.pipe(gulp.dest(config.dest + '/assets/toolkit/scripts'));
+gulp.task('scripts:toolkit', function (done) {
+	webpackCompiler.run(function (error, result) {
+		if (error) gutil.log(gutil.colors.red(error));
```
Comment (LukeAskew) [inline gulpfile.js]: style preference: always use curly braces on new lines. http://jshint.com/docs/options/#curly

`$ jshint gulpfile.js`

**Comment by LukeAskew** [comment]
Awesome :guitar:. Just some small code style things.

This new setup will affect the current recommendation for integrating third-party scripts. Is there a way to handle this in the `webpack.config.js`? Just needs to concat an array of files, in order, then concat **before** the `toolkit.js` bundle.

Optionally, some people might want to just concat third-party scripts into a separate file (e.g. `vendor.js`) and keep them decoupled from `toolkit.js`. Easy with webpack?

http://fbrctr.github.io/advanced/third-party.html

File: gulpfile.js (hunk 2 above)
Comment (aputinski) [inline gulpfile.js]: @LukeAskew I can move it, but it need to be below the config block. We could move the config to the very top?

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+var path = require('path');
+var webpack = require('webpack');
+
+module.exports = function(fabricatorConfig) {
+
+	var config = {
+		entry: {
+			'fabricator/scripts/f': fabricatorConfig.src.scripts.fabricator,
```
Comment (aputinski) [inline webpack.config.js]: I removed the `scripts:fabricator` task in favor of just bundling it with weback.

File: gulpfile.js [hunk 5]
```diff
@@ -43,6 +41,10 @@ var config = {
 };
 
 
+// webpack
+var webpackCompiler = webpack(require('./webpack.config')(config));
```
Comment (aputinski) [inline gulpfile.js]: This passes `config` to `webpack.config.js`, so it needs to be below the config.

**Comment by LukeAskew** [comment]
I'm getting some odd behavior when a script has syntax errors. The errors log to console  (vebose :+1: ) but the tasks stop re-running on file change. The browsers reload, but the tasks don't run.

**Comment by aputinski** [comment]
Hmm. I'll take a look. 






On Mon, Apr 13, 2015 at 5:33 AM -0700, "Luke Askew" <notifications@github.com> wrote:












I'm getting some odd behavior when a script has syntax errors. The errors log to console  (vebose  ) but the tasks stop re-running on file change. The browsers reload, but the tasks don't run.

**Pull request merged**